lib/plugins/sfDoctrinePlugin/lib/generator/sfDoctrineFormGenerator.class.php

```diff
@@ -554,44 +553,41 @@ public function underscore($name)
     return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), '\\1_\\2', $name));
   }
 
-  /**
-   * Get array of sfDoctrineColumn objects that exist on the current model but not its parent.
-   *
-   * @return array $columns
-   */
-  public function getColumns()
-  {
-    $parentModel = $this->getParentModel();
-    $parentColumns = $parentModel ? array_keys(Doctrine_Core::getTable($parentModel)->getColumns()) : array();
-
-    $columns = array();
-    foreach (array_diff(array_keys($this->table->getColumns()), $parentColumns) as $name)
+    /**
+     * Get array of sfDoctrineColumn objects that exist on the current model but not its parent.
+     *
+     * @return array $columns
+     */
+    public function getColumns()
     {
-      $columns[] = new sfDoctrineColumn($name, $this->table);
-    }
+        $parentModel = $this->getParentModel();
+        $parentColumns = $parentModel ? array_keys(Doctrine_Core::getTable($parentModel)->getColumns()) : [];
 
-    // add relations to columns for inherited classes
-    if ($parentModel)
-    {
-      $parentRelationNames = array_keys(Doctrine_Core::getTable($parentModel)->getRelations());
-      $relations = $this->table->getRelations();
-      $relationColumns = array();
-      foreach (array_diff(array_keys($relations), $parentRelationNames) as $relationName)
-      {
-        if (Doctrine_Relation::ONE == $relations[$relationName]->getType())
-        {
-          $columnName = $relations[$relationName]->getLocal();
-          if (!in_array($columnName, $relationColumns))
-          {
-            $relationColumns[] = $columnName;
-            $columns[] = new sfDoctrineColumn($columnName, $this->table);
-          }
+        $columns = [];
+        $names   = [];
+
+        foreach (array_diff(array_keys($this->table->getColumns()), $parentColumns) as $name) {
+            $columns[] = new sfDoctrineColumn($name, $this->table);
+            $names[]   = $name;
         }
-      }
-    }
 
-    return $columns;
-  }
+        // add relations to columns for inherited classes
+        if ($parentModel) {
+            $parentRelationNames = array_keys(Doctrine_Core::getTable($parentModel)->getRelations());
+            $relations = $this->table->getRelations();
+            foreach (array_diff(array_keys($relations), $parentRelationNames) as $relationName) {
+                if (Doctrine_Relation::ONE == $relations[$relationName]->getType()) {
+                    $columnName = $relations[$relationName]->getLocal();
+                    if (!in_array($columnName, $names)) {
+                        $columns[] = new sfDoctrineColumn($columnName, $this->table);
+                        $names[]   = $columnName;
+                    }
```

Comment:
...and we _don't_ attempt to add another relationship-based copy of that property if we've already added it previously.